src/include/column_inline.h

```diff
@@ -169,8 +201,15 @@ __col_insert_search(
     uint64_t ins_recno;
     int cmp, i;
 
+    /*
+     * CPU may reorder the reads and return a stale value of the tail.
+     *
+     * Place a read barrier here to avoid this issue.
+     */
+    WT_ORDERED_READ(ret_ins, WT_SKIP_LAST(ins_head));
+
     /* If there's no insert chain to search, we're done. */
-    if ((ret_ins = WT_SKIP_LAST(ins_head)) == NULL)
```

Comment:
I think in general, these could use some more comments. For example, it's not obvious here that we're protecting against `ins_stack[0] = &ret_ins->next[0]` being done "before" the null check. (At least, that's my understanding - and that we don't need to ordered reads in the loop since we're reading from `ins_head` which is local to our thread after the first case.)

Reply:
There is a single thread happen before relationship here between the read of ret_ins and the usage of the ret_ins to read  ret_ins->next[0]. I don't think we need to put a barrier to protect the reorder in this case. The issue is still compiler can replace the usage of ret_ins with another read.

Reply:
I think we should put that in a comment. It's not obvious to future maintainers why this barrier is here or how to correctly make modifications to this code - I'd prefer to get it all written down while the context is in our head. Same for the other comments above here, too.

Reply:
done